File: src/main/java/no/stunor/origo/eventorapi/model/firestore/Eventor.java
package no.stunor.origo.eventorapi.model.firestore;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.google.cloud.spring.data.firestore.Document;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collectionName = "eventors")
public class Eventor {
    private String eventorId;
    private String name;
    private String federation;
    private String baseUrl;
    @JsonIgnore
    private String apiKey;
}

File: src/main/java/no/stunor/origo/eventorapi/model/Eventor.kt
package no.stunor.origo.eventorapi.model

import com.fasterxml.jackson.annotation.JsonIgnore
import com.google.cloud.spring.data.firestore.Document
import java.io.Serializable

@Document(collectionName = "eventors")
data class Eventor (
    var eventorId: String = "",
    var name: String = "",
    var federation: String = "",
    var baseUrl: String = "",
    @JsonIgnore
    var apiKey: String? = null
) : Serializable